compiler/rustc_resolve/src/lib.rs

```diff
@@ -881,6 +881,10 @@ pub struct Resolver<'a> {
     /// Used for hints during error reporting.
     field_names: FxHashMap<DefId, Vec<Spanned<Symbol>>>,
 
+    /// Span of the privacy modifier in fields of an item `DefId` accessible with dot syntax.
+    /// Used for hints during error reporting.
+    field_visibility_spans: FxHashMap<DefId, Vec<Span>>,
+
```

Comment:
Would it make sense to extend and reuse the `field_names` field for this?

Reply:
`field_names` only had the `Symbol` and I wasn't sure if there was any place that relied on it being always a `Vec`, causing perf impact if needing to map/collect it every time. But yes, we could do that :)

Reply:
The only difference I see is that this approach will only store the privacy span vec for tuple structs and nothing else.